pgo/cmd/root.go

```diff
@@ -82,7 +84,18 @@ func initConfig() {
 		log.Debugf("using PGO_NAMESPACE env var %s", tmp)
 	}
 
-	GetCredentials()
+	// Get the pgouser and password information
+	SetSessionUserCredentials()
+
+	// If TLS is disabled, do not gather unneeded credentials
+	if PGO_DISABLE_TLS == true || os.Getenv("DISABLE_TLS") == "true" {
+		SetHTTPClient(nil)
+	} else {
+		// If TLS is enabled, gather credentials and supply them for the
+		// client connection
+		httpTransport := GetTLSCredentials()
+		SetHTTPClient(httpTransport)
+	}
```

Comment:
Suggestion:

```go
func initConfig() {
    var httpTransport http.Transport

    // blah...

    // If TLS is enabled, get the TLS credentials and set the httpTransport
    // object required to initialize the client
    if PGO_DISABLE_TLS != true && os.Getenv("DISABLE_TLS") != "true" {
        httpTransport = GetTLSCredentials()
    }

    SetHTTPClient(httpTransport)
}
```

(assuming I got DeMorgan's Law correct here)

Reply:
Updated to match, testing confirms the client still functions as expected.